### ISSUE TITLE
Difference between "number of examples" in logs and "number of examples per pass"

### ISSUE THREAD

**Issue by oleg131** [VowpalWabbit/vowpal_wabbit#3987]
VW API: Python wrapper
VW version: 8.10.1

I have ran VW twice with exact same parameters, but differently ordered data. The logs for both show slightly different average loss and the two models perform very differently on the test set. One difference I identified is that in the 1st job, the logs say

```
number of examples per pass = 53463030
passes used = 60
```
as it normally did for other (unrelated) models I ran, but in the 2nd job, they say
```
number of examples = 3226803390
```
Is there a significance to it? What could have caused this?

**Comment by jackgerrits** [comment]
Can you please try the same with the latest version to see if it behaves the same?

What are the parameters? Did you send that many examples to VW?

**Comment by oleg131** [comment]
Will try asap, currently not east for me to upgrade.

Full parameters: `--readable_model ./model.readable         --cache_file ./data.cache         -f /tmp/tmprwg122mq/model.final         --data /tmp/tmprwg122mq/train.vw         --classweight -1:0.168 --loss_function hinge --passes 60 --hash all --bit_precision 27 --l1 0.00000001 --ngram U2 -q ZU -q ZV -q ZW -q ZX -q ZY --learning_rate 1.0 --decay_learning_rate 0.7071068 --initial_t 1.0 --invariant --adaptive --normalized --power_t 0.1 --quantile_tau 0.5 --holdout_off`

Yeah, my full data size is around 58M rows but as you can see there are less rows that that per pass. I know there are 1027 duplicates but that wouldn't explain all of the difference.

**Comment by oleg131** [comment]
@jackgerrits
Do you think the model could have only done 1 pass but for some reason printed the number of examples in the pass being too large (times number of passes too large)?

**Comment by bassmang** [comment]
Hey @oleg131 
Based on the numbers you provided it looks like the numbers add up correctly (53463030 * 60 = 3226803390). Does the second job not use `--passes`?

**Comment by oleg131** [comment]
Both jobs uses `passes`. It turned out that the problem was in the cache file location, namely `--cache_file ./data.cache`

Seems that the model ran out of disk space but still completed training. However, there were many "Failed to write example" and "Truncated example" errors.

Once the cache file location was moved elsewhere with sufficient space, these errors disappeared and "number of examples per pass" was always the exact number of rows in data, and "passes used = 60" always appeared.
